pkg/shared/utils/sync/distributed_lock_mock.go

```diff
@@ -0,0 +1,39 @@
+package sync
+
+import (
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+)
+
+var _ DistributedLock = &distributedLockMock{}
+
+// It appears that go-mocket does not offer a straightforward method to intercept the Rollback call. To make it easily
+// interceptable, this mock substitutes the rollback call in the Unlock method with a DELETE.
+
+func newDistributedLockMock(db *gorm.DB, lockID string) DistributedLock {
+	dl := NewDistributedLock(db, lockID)
+	return &distributedLockMock{dl: dl, db: db, id: lockID}
+}
+
+type distributedLockMock struct {
```

Comment:
Does this work with a mocked database? My understanding is that that's the case
Maybe it would be good to explicitly mention that for potential users of the type. What do you think?